### PR TITLE
Add better erlang linter

### PR DESCRIPTION
**Pull request by urbanserj**

### CODE REVIEW

File: ale_linters/erlang/erlc.erl [hunk 1]
```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env escript
```
Comment (w0rp) [inline ale_linters/erlang/erlc.erl]: What is this and what does it even do? There aren't any comments anywhere.

Comment (urbanserj) [inline ale_linters/erlang/erlc.erl]: Sorry, I should’ve explained this earlier.

Erlang’s ecosystem doesn’t have a linker and each project has to implement it from scratch for its own needs (e.g. [1]). Erlc [2] is a compiler for Erlang, it knows nothing about build tooling that almost everyone uses (rebar3 [3][4]). You could think of it as gcc vs. cmake. It’s not possible to run rebar3 as a linter. So the script specifies paths for libraries and other compiler options.

[1] https://github.com/vim-erlang/vim-erlang-compiler/blob/master/compiler/erlang_check.erl
[2] http://erlang.org/doc/man/erlc.html
[3] http://www.rebar3.org
[4] https://github.com/erlang/rebar3

Comment (w0rp) [inline ale_linters/erlang/erlc.erl]: Could you add some comments to the Erlang script to explain this?

**Comment by w0rp** [comment]
I think you should add this as a new `rebar` linter, and then one linter or the other could be enabled or disabled as needed. Otherwise, changing the existing `erlc` linter will make it so some users who do not use `rebar3`, but do have a basic Erlang compiler, won't be able to check their files with ALE any more.

**Comment by urbanserj** [comment]
Should we ask an opinion of the initial author @evnu?

These linters have completely the same functionality. It’s the same compiler, but the script is setting up paths to dependencies and other options defined in the config file. It works the same in the case when your project doesn’t use rebar3 as a build tool. @w0rp, do you still insist on making it as the separate linter for ALE?

**Comment by evnu** [comment]
I like this implementation. My original implementation did not take `escript` into account, and "project support" using vim variables is probably a hazzle for people already using `rebar`. Note however that I'd still like the option to pass in custom parameters, as I for one do not use `rebar`. I usually use `localvimrc` to set up my project-specific configuration, which handles Erlang configuration for ale as well.

So, I'd suggest two changes to the linter presented here:

* Allow setting configuration with `g:ale_erlang_erlc_options` as before
* The module's file extension should be `.escript`

**Comment by evnu** [comment]
After having updated ale, I  see that it has another Erlang linter `syntaxerl` (see https://github.com/w0rp/ale/commit/1aea6a34ff22c9e79d20a018b4d46000c208caf2). Without having checked `syntaxerl`, do the proposed changes here overlap with `syntaxerl`?

**Comment by urbanserj** [comment]
Oh wow. I didn't know about this project! It works well for my needs. I think I can close my PR at the moment. Thank you very much!

**Pull request closed without merging**